// svix/src/openapi/models/AggregatedEventTypes.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

export class AggregatedEventTypes {
    /**
    * The Application\'s ID.
    */
    'appId': string;
    'explicitlySubscribedEventTypes': Array<string>;
    'hasCatchAllEndpoint': boolean;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "appId",
            "baseName": "appId",
            "type": "string",
            "format": ""
        },
        {
            "name": "explicitlySubscribedEventTypes",
            "baseName": "explicitlySubscribedEventTypes",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "hasCatchAllEndpoint",
            "baseName": "hasCatchAllEndpoint",
            "type": "boolean",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return AggregatedEventTypes.attributeTypeMap;
    }

    public constructor() {
    }
}
