// svix/src/openapi/models/ApplicationStatsOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { HttpFile } from '../http/http';

/**
* A copy of  [`backgroundtask::ApplicationStats`], but serialized with camelCase fields for customers.
*/
export class ApplicationStatsOut {
    /**
    * The Application\'s ID.
    */
    'appId': string;
    /**
    * The Application\'s UID.
    */
    'appUid'?: string | null;
    'messageDestinations': number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "appId",
            "baseName": "appId",
            "type": "string",
            "format": ""
        },
        {
            "name": "appUid",
            "baseName": "appUid",
            "type": "string",
            "format": ""
        },
        {
            "name": "messageDestinations",
            "baseName": "messageDestinations",
            "type": "number",
            "format": "int64"
        }    ];

    static getAttributeTypeMap() {
        return ApplicationStatsOut.attributeTypeMap;
    }

    public constructor() {
    }
}
