// svix/src/openapi/models/BackgroundTaskFinishedEvent.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { BackgroundTaskFinishedEvent2 } from '../models/BackgroundTaskFinishedEvent2';
import { HttpFile } from '../http/http';

/**
* Sent when a background task is finished.
*/
export class BackgroundTaskFinishedEvent {
    'data': BackgroundTaskFinishedEvent2;
    'type': BackgroundTaskFinishedEventTypeEnum;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "BackgroundTaskFinishedEvent2",
            "format": ""
        },
        {
            "name": "type",
            "baseName": "type",
            "type": "BackgroundTaskFinishedEventTypeEnum",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return BackgroundTaskFinishedEvent.attributeTypeMap;
    }

    public constructor() {
    }
}

export enum BackgroundTaskFinishedEventTypeEnum {
    BackgroundTaskFinished = 'background_task.finished'
}
