// svix/src/openapi/models/BackgroundTaskFinishedEvent2.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { BackgroundTaskStatus } from '../models/BackgroundTaskStatus';
import { BackgroundTaskType } from '../models/BackgroundTaskType';
import { Data } from '../models/Data';
import { HttpFile } from '../http/http';

export class BackgroundTaskFinishedEvent2 {
    'data': Data;
    'status': BackgroundTaskStatus;
    'task': BackgroundTaskType;
    /**
    * The QueueBackgroundTask\'s ID.
    */
    'taskId': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "data",
            "baseName": "data",
            "type": "Data",
            "format": ""
        },
        {
            "name": "status",
            "baseName": "status",
            "type": "BackgroundTaskStatus",
            "format": ""
        },
        {
            "name": "task",
            "baseName": "task",
            "type": "BackgroundTaskType",
            "format": ""
        },
        {
            "name": "taskId",
            "baseName": "taskId",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return BackgroundTaskFinishedEvent2.attributeTypeMap;
    }

    public constructor() {
    }
}
