// svix/src/openapi/models/DataAnyOf2.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { ApplicationMessageFailure } from '../models/ApplicationMessageFailure';
import { ApplicationMessageIdPair } from '../models/ApplicationMessageIdPair';
import { HttpFile } from '../http/http';

export class DataAnyOf2 {
    'messagesCreated': Array<ApplicationMessageIdPair>;
    'messagesFailed': Array<ApplicationMessageFailure>;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "messagesCreated",
            "baseName": "messages_created",
            "type": "Array<ApplicationMessageIdPair>",
            "format": ""
        },
        {
            "name": "messagesFailed",
            "baseName": "messages_failed",
            "type": "Array<ApplicationMessageFailure>",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return DataAnyOf2.attributeTypeMap;
    }

    public constructor() {
    }
}
