// svix/src/openapi/models/DataAnyOf3.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { AggregatedEventTypes } from '../models/AggregatedEventTypes';
import { HttpFile } from '../http/http';

export class DataAnyOf3 {
    'eventTypes': Array<AggregatedEventTypes>;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "eventTypes",
            "baseName": "event_types",
            "type": "Array<AggregatedEventTypes>",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return DataAnyOf3.attributeTypeMap;
    }

    public constructor() {
    }
}
