// svix/src/openapi/models/DataAnyOf1.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { ApplicationStatsOut } from '../models/ApplicationStatsOut';
import { HttpFile } from '../http/http';

export class DataAnyOf1 {
    'appStats': Array<ApplicationStatsOut>;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "appStats",
            "baseName": "app_stats",
            "type": "Array<ApplicationStatsOut>",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return DataAnyOf1.attributeTypeMap;
    }

    public constructor() {
    }
}
