// svix/src/openapi/models/Data.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { AggregatedEventTypes } from '../models/AggregatedEventTypes';
import { ApplicationMessageFailure } from '../models/ApplicationMessageFailure';
import { ApplicationMessageIdPair } from '../models/ApplicationMessageIdPair';
import { ApplicationStatsOut } from '../models/ApplicationStatsOut';
import { DataAnyOf } from '../models/DataAnyOf';
import { DataAnyOf1 } from '../models/DataAnyOf1';
import { DataAnyOf2 } from '../models/DataAnyOf2';
import { DataAnyOf3 } from '../models/DataAnyOf3';
import { DataAnyOf4 } from '../models/DataAnyOf4';
import { HttpFile } from '../http/http';

/**
* A copy of [`backgroundtask::Data`], but serialized with camelCase fields for customers.
*/
export class Data {
    'messagesSent': number;
    'appStats': Array<ApplicationStatsOut>;
    'messagesCreated': Array<ApplicationMessageIdPair>;
    'messagesFailed': Array<ApplicationMessageFailure>;
    'eventTypes': Array<AggregatedEventTypes>;
    'messagesPurged': number;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "messagesSent",
            "baseName": "messages_sent",
            "type": "number",
            "format": "uint"
        },
        {
            "name": "appStats",
            "baseName": "app_stats",
            "type": "Array<ApplicationStatsOut>",
            "format": ""
        },
        {
            "name": "messagesCreated",
            "baseName": "messages_created",
            "type": "Array<ApplicationMessageIdPair>",
            "format": ""
        },
        {
            "name": "messagesFailed",
            "baseName": "messages_failed",
            "type": "Array<ApplicationMessageFailure>",
            "format": ""
        },
        {
            "name": "eventTypes",
            "baseName": "event_types",
            "type": "Array<AggregatedEventTypes>",
            "format": ""
        },
        {
            "name": "messagesPurged",
            "baseName": "messages_purged",
            "type": "number",
            "format": "uint"
        }    ];

    static getAttributeTypeMap() {
        return Data.attributeTypeMap;
    }

    public constructor() {
    }
}
