// svix/src/openapi/models/EndpointOauthConfigOut.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { Oauth2AuthMethodInOut } from '../models/Oauth2AuthMethodInOut';
import { Oauth2GrantTypeInOut } from '../models/Oauth2GrantTypeInOut';
import { HttpFile } from '../http/http';

export class EndpointOauthConfigOut {
    'authMethod': Oauth2AuthMethodInOut;
    'clientId': string;
    'extraParams'?: { [key: string]: string; } | null;
    'grantType': Oauth2GrantTypeInOut;
    'scopes'?: Array<string> | null;
    'tokenUrl': string;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "authMethod",
            "baseName": "authMethod",
            "type": "Oauth2AuthMethodInOut",
            "format": ""
        },
        {
            "name": "clientId",
            "baseName": "clientId",
            "type": "string",
            "format": ""
        },
        {
            "name": "extraParams",
            "baseName": "extraParams",
            "type": "{ [key: string]: string; }",
            "format": ""
        },
        {
            "name": "grantType",
            "baseName": "grantType",
            "type": "Oauth2GrantTypeInOut",
            "format": ""
        },
        {
            "name": "scopes",
            "baseName": "scopes",
            "type": "Array<string>",
            "format": ""
        },
        {
            "name": "tokenUrl",
            "baseName": "tokenUrl",
            "type": "string",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return EndpointOauthConfigOut.attributeTypeMap;
    }

    public constructor() {
    }
}
