// svix/src/openapi/models/EnvironmentIn.ts
/**
 * Svix API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * OpenAPI spec version: 1.1.1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { ConnectorIn } from '../models/ConnectorIn';
import { EventTypeIn } from '../models/EventTypeIn';
import { HttpFile } from '../http/http';

export class EnvironmentIn {
    'connectors'?: Array<ConnectorIn> | null;
    'eventTypes'?: Array<EventTypeIn> | null;
    'settings'?: any | null;

    static readonly discriminator: string | undefined = undefined;

    static readonly mapping: {[index: string]: string} | undefined = undefined;

    static readonly attributeTypeMap: Array<{name: string, baseName: string, type: string, format: string}> = [
        {
            "name": "connectors",
            "baseName": "connectors",
            "type": "Array<ConnectorIn>",
            "format": ""
        },
        {
            "name": "eventTypes",
            "baseName": "eventTypes",
            "type": "Array<EventTypeIn>",
            "format": ""
        },
        {
            "name": "settings",
            "baseName": "settings",
            "type": "any",
            "format": ""
        }    ];

    static getAttributeTypeMap() {
        return EnvironmentIn.attributeTypeMap;
    }

    public constructor() {
    }
}
